client/executor/src/wasm_runtime.rs

```diff
@@ -62,7 +62,7 @@ struct VersionedRuntime {
 	/// Runtime version according to `Core_version` if any.
 	version: Option<RuntimeVersion>,
 	/// Cached instance pool.
-	instances: RwLock<[Option<Arc<Mutex<Box<dyn WasmInstance>>>>; MAX_INSTANCES]>,
+	instances: RwLock<Vec<Option<Arc<Mutex<Box<dyn WasmInstance>>>>>>,
```

Comment:
I used an array here to avoid an allocation for each call when this is cloned. Unfortunately with `parking_lot` I could not find a way to `try_lock` a mutex and return a locked guard for an clone of the `Arc<Mutex>`. An entire set of instances needs to be cloned so that the `RwLock` can be released for the duration of the call.
Consider using `SmallVec` here instead.

Also, with `Vec` or `SmallVec` you can use `RwLock<Vec<Arc<..` instead of `RwLock<Vec<Option<Arc<..`. Simply allocate a new instance by pushing it to the vector.